_posts/2019-07-10-quarkus-0-19-released.adoc

```diff
@@ -0,0 +1,101 @@
+---
+layout: post
+title: 'Quarkus 0.19.0 released'
+date: 2019-07-10
+tags: quarkus release
+author: gsmet
+---
+
+Quarkus 0.19.0 was just released and is now available on Maven Central.
+
+This new version brings a lot of new extensions, especially in the persistence area.
+
+[NOTE]
+====
+Quarkus 0.19.0 is still targeting GraalVM 19.0.2 as we discovered a bug in GraalVM 19.1.0 that affects Quarkus.
+
+The issue is already fixed but we are waiting for a 19.1.1 release.
+====
+
+== What's new?
+
+Here are the main highlights of this release.
+
+=== Hibernate ORM schema management improvements
+
+We had a few issues with automatic schema upgrades handled by Hibernate ORM.
+
+Things are now smoother and we significantly improved the documentation to explain TODO link[how to configure schema management in various situations].
+
+=== New extensions
+
+==== MongoDB extension
+
+https://www.mongodb.com/[MongoDB] needs no introduction as it's probably the most widely used NoSQL datastore.
+We had a lot of users asking for MongoDB support in Quarkus and here we are: we now have a fully featured MongoDB client extension.
+
+It supports imperative and reactive and comes with a lot of configuration properties to tune the client.
+
+And, of course, you can use it with GraalVM.
+
+You can learn more about it in the link:/guides/mongo-guide[MongoDB client guide].
+
+==== Neo4j extension
+
+https://neo4j.com/[Neo4j] is a graph database management system focused not only on the data itself, but especially on the relations between data.
+
+Neo4j offers Cypher, a declarative query language much like SQL.
+Cypher is used to query the graph, telling the database engine what patterns to match and how.
+Clients communicate over the so called Bolt protocol with the database.
+
+Quarkus now offers a Neo4j extension so that you can easily use Neo4j with Quarkus in JVM and native mode.
+
+A guide is coming soon but you can already take a look at https://github.com/quarkusio/quarkus/tree/master/integration-tests/neo4j[the integration tests] if you can't wait for it.
+
+==== Amazon DynamoDB extension
```

Comment:
I am curious why we let these not documented and not really configuration ready extensions in.

Reply:
I think it's an interesting contribution that can already be useful for early adopters (if you want to connect to Amazon DynamoDB in native mode).

We already have a follow-up PR to make progress in the right direction.

That's not something I would do post 1.0 but I think it's a good way to keep contributors with us and attract new ones, while we are still evolving quickly.

It's also related to a discussion we had about the platform coming and how to deal with externally contributed extensions.

Reply:
BTW, just to give a bit more context on these particular extensions, the authors are actively working on documentation/quarkification and so on.

They haven't dumped the extensions and run :).

See https://github.com/quarkusio/quarkus/pull/3171 or https://github.com/quarkusio/quarkus/pull/3172